### ISSUE TITLE
CLI command to remove all old environments

### ISSUE THREAD

**Issue by zachvalenta** [python-poetry/poetry#1884]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate. -> https://github.com/python-poetry/poetry/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+removing+old+environments
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered. --> https://python-poetry.org/docs/managing-environments/#deleting-the-environments

## Feature Request

__tldr__: Is there a way to way to clean up old environments en masse?

__context__: I often spin up a quick project to explore an idea and then tear it down once I'm past the exploration phase. Using `pip`, the environment could be co-located to the project i.e. `rm <my_project>` would also remove the environment. With Poetry, environments are located outside the project, and thus it's possible to delete the project without deleting the environment. As a result, you could have environments at `path/to/pypoetry/virtualenvs/` whose projects have been deleted. Currently, Poetry only offers the ability to remove an environment using the CLI from within the project itself; if you've already deleted the project, you need to navigate to `path/to/pypoetry/virtualenvs/` and manually remove its hanging environment. It'd be nice if Poetry had a CLI command in which you could remove all environments that no longer correspond to source code directories on the operating system.

**Comment by finswimmer** [comment]
Hello,

you can [config](https://python-poetry.org/docs/configuration/#virtualenvsin-project-boolean) poetry to put the venv within the project using `virtualenvs.in-project 
 true`.

I don't think it is possible to remove venv where the project does not exists anymore via CLI, because there is no information about where the location of the project was.

Having a command to cleanup all venv for an existing project should be possible and seems quite useful to me.

fin swimmer

**Comment by jtrakk** [comment]
There could be an option


```
poetry env remove --all
```
or it could be

```
poetry env remove one python3.6
poetry env remove all
```

**Comment by abn** [comment]
Resolved-by: #3212

**Comment by am1006** [comment]
> Resolved-by: #3212

Not sure if this was answered before. After trial and error, I don't quite get how this merge could fix the problem that OP tried to fix. From #3212 I can see that the `-all` argument can remove all environments that are attached to the current project. Could you advise me how this will remove all hanging environments (i.e., the problem that OP raised)?

**Comment by DominikRoB** [comment]
Found this here, because I have the same issue as OP.
Is it fine to delete environments (hanging or not) from path/to/pypoetry/virtualenvs/ or are there negative consequences?

**Comment by vpunch** [comment]
> Resolved-by: #3212

It is not a solution to the problem described by OP. I also find it strange that poetry does not delete virtual environments after deleting the directory with the project.
